objects/69ead2/contained/DUESETUP/script.lua

```diff
@@ -0,0 +1,643 @@
+    end
+
+    if setupComplete then
+        createPanelButtons()
+        createAllCardButtons()
+    end
+end
+
+function null()
+end
+
+function doSetup(params)
+    local color = params.color
+    local panel = Global.call("getSpirit", {name = spiritName})
+    self.locked = true
+    self.addTag("Modify Cost")
+    local position = panel.getPosition() + Vector(-8.2,-0.22,6.8)
+    self.setPosition(position)
+    
+    -- hand for impending cards:
+    local position1 = Player[color].getHandTransform(2).position
+    position1.z = position1.z - 5.5
+    Global.call("SpawnHand", {color = color, position = position1})
+    
+    Wait.time(function() createPanelButtons() end, 0.1) -- without a wait, if spirit chosen after setup, buttons don't appear
+    
+    return true
+end
+
+-----------------------------------------
+-- buttons and functions for spirit panel
+-----------------------------------------
+
+function createPanelButtons()
+    local dances = Global.call("getSpirit", {name = spiritName})
+    dances.createButton({
+        click_function = "impendCard",
+        function_owner = self,
+        label = "Impend a card!",
+        position = {-0.93,0.5,0},
+        scale = {x=0.08, y=0.08, z=0.08},
+        width = 5000,
+        height = 750,
+        font_size = 750,
+        color = {1,1,1},
+        font_color = {0,0,0},
+        tooltip = "Place a card on the panel art\nand press the button to impend it!",
+    })
+    
+    dances.createButton({
+        click_function = "g3Toggle",
+        function_owner = self,
+        label = "☐\n\n\n", 
+        position = {1.1,0.2,-0.78},
+        scale = {x=0.08, y=0.08, z=0.08},
+        width = 1800,
+        height = 1800,
+        font_size = 450,
+        color = {0,0,0,0.5},
+        font_color = {0,0,0,100},
+        tooltip = "Press this if you are choosing the third growth option in order to show buttons on impending cards.",
+    })
+    if g3Selected then
+        local ind = getButtonIndex(dances, "g3Toggle")
+        dances.editButton({
+            index = ind,
+            label = "☑\n\n\n",
+        })
+    end
+end
+
+function impendCard(obj)
+    local color = Global.call("getSpiritColor", {name = spiritName})
+    local hits = Physics.cast({
+        origin = obj.getPosition() + Vector(-5,0,2),
+        direction = Vector(0,1,0),
+        type = 3,
+        size = {6,1,6.5},
+        max_distance = 1,
+        -- debug = true
+    })
+    for _,hit in pairs(hits) do
+        if hit.hit_object.type == "Card" and hit.hit_object.getVar("energy") ~= nil then -- check cards for energy to make sure they are correct hits
+            local card = hit.hit_object
+            if impendTable[card.guid] == nil then
+                local costs = {}
+                costs[card.guid] = card.getVar("energy")
+                local energy = Global.call("modifyCost",{color = color, costs = costs})[card.guid]
+                card.hide_when_face_down = false
```

Comment:
There's override buttons on the back of the card (which I intentionally placed on the back to keep out of the way as they shouldn't be needed often but will come up frequently enough to need them) and if the power card has this set to true, these buttons do not show while the card is facing down. If you have alternate ideas, I'm open to suggestion since this is a little ugly.

Reply:
hmm you could do a context menu item, but there's no way to remove one individually, so maybe not best there. i'm trying to remember why i had this set to true for power cards, i think it was for `Diversity of Spirits`?